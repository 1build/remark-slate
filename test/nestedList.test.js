const plugin = require('../');
const nestedMdast = require('./nested-mdast');

const transform = plugin.transform;

it('When flattenListItems is false, the list has paragraph elements', () => {
  expect(nestedMdast.children.map(transform)).toMatchSnapshot();
});

it('When flattenListItems is true, the list-items have leaf nodes', () => {
  expect(
    nestedMdast.children.map((c) => transform(c, { flattenListItems: true }))
  ).toMatchSnapshot();
});
