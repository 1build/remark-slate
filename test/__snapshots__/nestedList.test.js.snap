// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When flattenListItems is false, the list has paragraph elements 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Bullet 1",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Bullet 2",
              },
            ],
            "type": "paragraph",
          },
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "text": "Nested 1 Bullet 1",
                      },
                    ],
                    "type": "paragraph",
                  },
                ],
                "type": "list-item",
              },
              Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "text": "Nested 1 Bullet 2",
                      },
                    ],
                    "type": "paragraph",
                  },
                  Object {
                    "children": Array [
                      Object {
                        "children": Array [
                          Object {
                            "children": Array [
                              Object {
                                "text": "Nested 2 Bullet 1",
                              },
                            ],
                            "type": "paragraph",
                          },
                        ],
                        "type": "list-item",
                      },
                      Object {
                        "children": Array [
                          Object {
                            "children": Array [
                              Object {
                                "text": "Nested 2 Bullet 2",
                              },
                            ],
                            "type": "paragraph",
                          },
                        ],
                        "type": "list-item",
                      },
                    ],
                    "type": "bulleted-list",
                  },
                ],
                "type": "list-item",
              },
            ],
            "type": "bulleted-list",
          },
        ],
        "type": "list-item",
      },
    ],
    "type": "bulleted-list",
  },
]
`;

exports[`When flattenListItems is true, the list-items have leaf nodes 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "text": "Bullet 1",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "text": "Bullet 2",
          },
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "text": "Nested 1 Bullet 1",
                  },
                ],
                "type": "list-item",
              },
              Object {
                "children": Array [
                  Object {
                    "text": "Nested 1 Bullet 2",
                  },
                  Object {
                    "children": Array [
                      Object {
                        "children": Array [
                          Object {
                            "text": "Nested 2 Bullet 1",
                          },
                        ],
                        "type": "list-item",
                      },
                      Object {
                        "children": Array [
                          Object {
                            "text": "Nested 2 Bullet 2",
                          },
                        ],
                        "type": "list-item",
                      },
                    ],
                    "type": "bulleted-list",
                  },
                ],
                "type": "list-item",
              },
            ],
            "type": "bulleted-list",
          },
        ],
        "type": "list-item",
      },
    ],
    "type": "bulleted-list",
  },
]
`;
